services/src/main/java/org/keycloak/keys/ImportedRsaKeyProviderFactory.java

```diff
@@ -84,6 +88,35 @@ public void validateConfiguration(KeycloakSession session, RealmModel realm, Com
             if (!certificate.getPublicKey().equals(keyPair.getPublic())) {
                 throw new ComponentValidationException("Certificate does not match private key");
             }
+            if (model.contains(Attributes.CERTIFICATE_CHAIN_KEY)) {
+                List<X509Certificate> certificateChain;
+                try {
+                    certificateChain = PemUtils.decodeCertificateChain(model.get(Attributes.CERTIFICATE_CHAIN_KEY));
+                } catch (Throwable t) {
+                    throw new ComponentValidationException("Failed to decode certificate chain", t);
+                }
+
+                if (certificateChain.isEmpty()) {
+                    throw new ComponentValidationException("Failed to decode certificate chain");
+                }
+
+                if (!certificate.equals(certificateChain.get(0))) {
+                    throw new ComponentValidationException("Server certificate must match first certificate in chain");
+                }
+
+                try {
+                    final KeyStore truststore = session.getProvider(TruststoreProvider.class).getTruststore();
+                    PKIXParameters params = new PKIXParameters(truststore);
+                    params.setRevocationEnabled(false);
```

Comment:
I'm assuming here that it is valid to have a root certificate in the certificate chain.
Related: https://stackoverflow.com/a/38128883/11347438